Userland/Services/NotificationServer/NotificationWindow.cpp

```diff
@@ -60,14 +62,28 @@ void update_notification_window_locations()
 
 NotificationWindow::NotificationWindow(const String& text, const String& title, const Gfx::ShareableBitmap& icon)
 {
-    s_windows.append(this);
+    while (true) {
+        m_window_id = get_random<i64>();
+        if (m_window_id < 0) {
+            if (m_window_id == (i64)0xFFFFFFFFFFFFFFFF) {
+                m_window_id = 0;
+            } else {
+                m_window_id = -m_window_id;
```

Comment:
Hmm? Why not use `u64` directly? It seems like the only place that uses `-1` as a special value is `GUI::Notification`, and there it is redundant to `m_showing`. Allowing `-1` (or switching to `u64`) would get rid of all the special cases.

Reply:
Changed to the client id with the lifetime connection, should never rollover (unless you wanna make 4 billion notifications) now :)